compiler/native/environment.go

```diff
@@ -36,15 +36,22 @@ func (c *client) EnvironmentStages(s yaml.StageSlice) (yaml.StageSlice, error) {
 func (c *client) EnvironmentSteps(s yaml.StepSlice) (yaml.StepSlice, error) {
 	// iterate through all steps
 	for _, step := range s {
+		env := make(map[string]string)
 		// gather set of default environment variables
-		env := environment(c.build, c.metadata, c.repo, c.user)
+		defaultEnv := environment(c.build, c.metadata, c.repo, c.user)
 
 		// inject the declared environment
 		// variables to the build step
 		for k, v := range step.Environment {
 			env[k] = v
 		}
 
+		// inject the default environment
+		// variables to the build step
```

Comment:
i think it might be nice to call out why we are doing this here, ie. ensure that default env vars are not overwritten.

Reply:
Added comments to hopefully clarify this.